cvmfs/swissknife_lease_curl.cc

```diff
@@ -0,0 +1,99 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include "swissknife_lease_curl.h"
+
+#include <iostream>
+
+size_t RecvCB(void *buffer, size_t size, size_t nmemb, void *userp) {
+  CurlBuffer* my_buffer = static_cast<CurlBuffer*>(userp);
+
+  if (size * nmemb < 1) {
+    return 0;
+  }
+
+  my_buffer->data = static_cast<char*>(buffer);
+
+  return my_buffer->data.size();
+}
+
+CURL* PrepareCurl(const char* method) {
+  CURL* h_curl = curl_easy_init();
+
+  if (h_curl) {
+    curl_easy_setopt(h_curl, CURLOPT_NOPROGRESS, 1L);
+    curl_easy_setopt(h_curl, CURLOPT_USERAGENT, "curl/7.47.0");
+    curl_easy_setopt(h_curl, CURLOPT_MAXREDIRS, 50L);
+    curl_easy_setopt(h_curl, CURLOPT_CUSTOMREQUEST, method);
+    curl_easy_setopt(h_curl, CURLOPT_TCP_KEEPALIVE, 1L);
+  }
+
+  return h_curl;
+}
+
+bool MakeAcquireRequest(const std::string& user_name,
+                        const std::string& lease_fqdn,
+                        const std::string& repo_service_url,
+                        CurlBuffer* buffer) {
+  CURLcode ret = static_cast<CURLcode>(0);
+
+  CURL* h_curl = PrepareCurl("POST");
+  if (h_curl) {
+    // Prepare payload
+    std::string payload = "{\"user\" : \"" + user_name +
+        "\", \"path\" : \"" + lease_fqdn + "\"}";
+
+    // Make request to acquire lease from repo services
+    curl_easy_setopt(h_curl,
+                     CURLOPT_URL,
+                     (repo_service_url + "/api/leases").c_str());
```

Comment:
Makes sense. I'll put this in a separate PR in sync with one in the services repo.